posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+
+== Installing and configuring Jenkins to setup CI/CD of Jakarta EE & MicroProfile application
+
+Install *jenkins* with all the recommended plug-ins link:https://www.jenkins.io/doc/book/installing/[install Jenkins], make sure that the following plug-ins are installed. You can install Jenkins on any physical or virtual servers, or it can be running as a container on Kubernetes itself.
+
+* Maven
+* Pipeline
+* Multibranch Scan Webhook Trigger
+* Docker
+* Kubernetes
+
+Jenkins builds the Jakarta EE & MicroProfile application code by using Jenkins pipeline scripts. The script can run directly on your Jenkins installation (known as Jenkins master) but, if the application is big, you need to use a Jenkins slave agent (good configuration server (or container)). More details about Jenkins slave setup are provided at link:https://www.jenkins.io/doc/book/using/using-agents/[Jenkins slave setup], link:https://www.jenkins.io/doc/book/pipeline/syntax/[pipeline code details] 
+
+== Jenkins pipeline scripting introduction
+
+It is recommended to adhere to the concept of infrastructure as a code (IaaC) in DevOps pipeline, pipeline scripting-based jenkins jobs are a good example of IaaC.
+
+You can write link:https://www.jenkins.io/doc/book/pipeline/syntax/[Jenkins pipeline code] in one of the following ways:
+
+* Directly in the link:https://www.jenkins.io/doc/book/pipeline/getting-started/[Jenkins UI]. This is a useful way to get started. You would need to take a backup of the Jenkins instance to store the pipeline code because it is part of the Jenkins installation. 
+
+image::/img/blog/pipeline-code-on-jenkins.png[Pipeline code directly on Jenkins ,width=70%,align="center"]
+
+* Jenkinsfile (a plain text file) in Git and mapping it to Jenkins. This is better for ensuring that your configuration is always under version control. If you have specific build and deployment configurations for separate environments, such as dev, staging, and production, you create a separate Jenkinsfile for each environment and store it in that environment specific Git repo  branches. You can either use "Pipeline" type job or "Multibranch pipeline" type job in this case.
+
+image::/img/blog/pipeline-code-on-git.png[Pipeline code on Git ,width=50%,align="center"]
+
+== Building the Jakarta EE & MicroProfile application with Open Liberty on Jenkins
+
+Before writing pipeline script to automate the deployment, you need to make a change in Dockerfile. The standard Dockerfile that you use in general may not enough to do the Jakarta EE & MicroProfile application deployment (to OpenLiberty server) on containerized environment. A sample Openliberty compliant Dockerfile snipped in given, you can configure it as needed (gather the additional files mentioned in this code from open liberty support team).
```

Comment:
What's changed in this sample? It's not clear from this explanation what needs changing. Couldn't we keep it simple here and just say to use this (greatly simplified) Dockerfile?

(the text in this explanation needs tidying up for grammar and capitalisation but I'm not sure what it needs to say so I'll not do that just now)

Reply:
Removed everything and kept only "Build your Dockerfile from the foundation code given below (gather the additional files mentioned in this code from open liberty support team)."